src/Drupal/Commands/core/LocaleCommands.php

```diff
@@ -128,4 +133,168 @@ public function update($options = ['langcodes' => self::REQ])
+     *   Export the Dutch customized and not customized translations.
+     * @usage drush locale:export --template > drupal.pot
+     *   Export the template file to translate.
+     * @aliases locale-export
+     * @see \Drupal\locale\Form\ExportForm::submitForm
+     *
+     * @throws \Exception
+     */
+    public function export($options = ['template' => false, 'langcode' => self::OPT, 'types' => self::OPT])
+    {
+        $language = $this->getTranslatableLanguage($options['langcode']);
+        $poreader_options = $this->getReaderOptions(StringUtils::csvToArray($options['types']));
+
+        $file_uri = drush_save_data_to_temp_file('temporary://', 'po_');
+        if ($this->writePoFile($file_uri, $language, $poreader_options)) {
+            $this->printFile($file_uri);
+        } else {
+            $this->logger()->notice(dt('Nothing to export.'));
+        }
+    }
+
+    /**
+     * Assure that required options are set.
+     *
+     * @hook validate locale:export
+     */
+    public function exportValidate(CommandData $commandData)
+    {
+        $langcode = $commandData->input()->getOption('langcode');
+        $template = $commandData->input()->getOption('template');
+
+        if (!$langcode && !$template) {
+            throw new \Exception(dt('Set --langcode=LANGCODE or --template, see help for more information.'));
+        }
+    }
+
+    /**
+     * Get translatable language object.
+     *
+     * @param string $langcode The language code of the language object.
+     * @return LanguageInterface|null
+     * @throws \Exception
+     */
+    private function getTranslatableLanguage($langcode)
+    {
+        if (!$langcode) {
+            return null;
+        }
+
+        $language = \Drupal::languageManager()->getLanguage($langcode);
+
+        if (!$language) {
+            throw new \Exception(dt('Language code @langcode is not configured.', [
+                '@langcode' => $langcode,
+            ]));
+        }
+
+        if (!$this->isTranslatable($language)) {
+            throw new \Exception(dt('Language code @langcode is not translatable.', [
+                '@langcode' => $langcode,
+            ]));
+        }
+
+        return $language;
+    }
+
+    /**
+     * Check if language is translatable.
+     *
+     * @param LanguageInterface $language
+     * @return bool
+     */
+    private function isTranslatable(LanguageInterface $language)
+    {
+        if ($language->isLocked()) {
+            return false;
+        }
+
+        if ($language->getId() != 'en') {
+            return true;
+        }
+
+        return (bool)\Drupal::config('locale.settings')->get('translate_english');
+    }
+
+    /**
+     * Get PODatabaseReader options.
+     *
+     * @param array $types
+     * @return array
+     * @throws \Exception
+     */
+    private function getReaderOptions(array $types = [])
+    {
+        $allowed_types = [
```

Comment:
Either $allowed_types or $types can use some documentation. Allowed types 'key' is type accepted by Reader, 'value' is drush command option input. Without comments, I have to read and interpret the relatively complex last two lines of this method.